<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tutorial 3: Human PBMCs (ATAC + RNA + Protein Epitopes) • mTopic</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial 3: Human PBMCs (ATAC + RNA + Protein Epitopes)">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mTopic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tutorial-1-data-preparation.html">Data Preparation</a>
    </li>
    <li>
      <a href="../articles/tutorial-2-human-tonsil.html">Human Tonsil (RNA + Protein Epitopes)</a>
    </li>
    <li>
      <a href="../articles/tutorial-3-human-pbmcs.html">Human PBMCs (ATAC + RNA + Protein Epitopes)</a>
    </li>
    <li>
      <a href="../articles/tutorial-4-P22-mouse-brain.html">P22 Mouse Brain (ATAC + RNA)</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">API</a>
</li>
<li>
  <a href="https://github.com/TabakaLab/mTopic" class="external-link">Python Package</a>
</li>
<li>
  <a href="https://github.com/TabakaLab/mTopic/tree/main/dashboard" class="external-link">Interactive Dashboard</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/TabakaLab/mTopicR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Tutorial 3: Human PBMCs (ATAC + RNA + Protein
Epitopes)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TabakaLab/mTopicR/blob/HEAD/vignettes/tutorial-3-human-pbmcs.Rmd" class="external-link"><code>vignettes/tutorial-3-human-pbmcs.Rmd</code></a></small>
      <div class="hidden name"><code>tutorial-3-human-pbmcs.Rmd</code></div>

    </div>

    
    
<p>Welcome to this tutorial on using the <code>mTopic</code> package for
non-spatial multimodal topic modeling of human peripheral blood
mononuclear cells (PBMCs), using the <a href="https://www.nature.com/articles/s41587-021-00927-2" class="external-link">DOGMA-seq
dataset</a>.</p>
<p>In this tutorial, we will cover the following steps:</p>
<ul>
<li>Scaling and normalizing the data.</li>
<li>Applying non-spatial multimodal topic model to identify distinct
cell populations and explore their functional roles.</li>
<li>Visualizing the results to gain insights into the composition and
structure of the immune cell landscape.</li>
</ul>
<p>Let’s begin by importing the required libraries and setting up the
working directory for this analysis. The dataset is available at <a href="https://zenodo.org/records/15431784" class="external-link">Zenodo</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TabakaLab/mTopicR" class="external-link">mTopic</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">dataset_name</span> <span class="op">&lt;-</span> <span class="st">"Human_PBMC_ATAC_RNA_Protein_filtered.rds"</span></span>
<span><span class="va">working_dir</span> <span class="op">&lt;-</span> <span class="st">'~/projects/datasets/'</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="va">dataset_name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span></span>
<span>    url <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/15431784/files/"</span>, <span class="va">dataset_name</span>, <span class="st">"?download=1"</span><span class="op">)</span>, </span>
<span>    destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="va">dataset_name</span><span class="op">)</span>, </span>
<span>    mode <span class="op">=</span> <span class="st">"wb"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="section level2">
<h2 id="non-spatial-multimodal-topic-modeling">Non-Spatial Multimodal Topic Modeling<a class="anchor" aria-label="anchor" href="#non-spatial-multimodal-topic-modeling"></a>
</h2>
<p>The filtered dataset contains 7,349 cells and includes three
modalities:</p>
<ul>
<li>
<code>atac</code>: chromatin accessibility data (Assay for
Transposase-Accessible Chromatin),</li>
<li>
<code>rna</code>: gene expression data,</li>
<li>
<code>prot</code>: protein abundance data.</li>
</ul>
<p>This rich multimodal structure enables integrated modeling of
transcriptional activity, regulatory accessibility, and surface protein
expression at single-cell resolution.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="va">dataset_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pbmc</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 147123 features across 7349 samples within 3 assays </span></span>
<span><span class="co">#&gt; Active assay: atac (136915 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  1 layer present: counts</span></span>
<span><span class="co">#&gt;  2 other assays present: rna, prot</span></span></code></pre></div>
<p>To prepare the dataset for topic modeling (as in Tutorial 2), we need
to transform and scale the raw counts across all modalities:</p>
<ul>
<li>TF-IDF transformation (<code>tfidf</code>) for <code>atac</code> and
<code>rna</code>. Emphasizes informative, modality-specific features by
balancing frequency and importance.</li>
<li>CLR normalization (<code>clr</code>) for <code>prot</code>. Corrects
compositional bias in protein counts across cells.</li>
</ul>
<p>After normalization, apply <code>scale_counts</code> to scale total
counts across modalities, ensuring balanced contribution to the topic
model.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tfidf.html">tfidf</a></span><span class="op">(</span><span class="va">pbmc</span>, mod <span class="op">=</span> <span class="st">'atac'</span><span class="op">)</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tfidf.html">tfidf</a></span><span class="op">(</span><span class="va">pbmc</span>, mod <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clr.html">clr</a></span><span class="op">(</span><span class="va">pbmc</span>, mod <span class="op">=</span> <span class="st">'prot'</span><span class="op">)</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scale_counts.html">scale_counts</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span></span></code></pre></div>
<p>To train a non-spatial topic model using <code>MTM</code>, define the
model, run Variational Inference (<code>VI</code>) for 500 iterations,
and export the trained parameters to the <code>pbmc</code> object.</p>
<p>While we use 100 iterations in this tutorial for thorough training,
the model often converges to meaningful topics in as few as 20
iterations. You can adjust the number of iterations based on dataset
size and desired precision.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mtm_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MTM.html">MTM</a></span><span class="op">(</span><span class="va">pbmc</span>, n_topics <span class="op">=</span> <span class="fl">30</span>, n_jobs <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">mtm_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/VI.html">VI</a></span><span class="op">(</span><span class="va">mtm_model</span>, n_iter <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/export_params.html">export_params</a></span><span class="op">(</span><span class="va">mtm_model</span>, <span class="va">pbmc</span><span class="op">)</span></span>
<span><span class="va">pbmc</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 147123 features across 7349 samples within 3 assays </span></span>
<span><span class="co">#&gt; Active assay: atac (136915 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  2 other assays present: rna, prot</span></span>
<span><span class="co">#&gt;  1 dimensional reduction calculated: mTopic</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualizing-results">Visualizing Results<a class="anchor" aria-label="anchor" href="#visualizing-results"></a>
</h2>
<p>To visualize the MTM results, compute a 2-dimensional UMAP embedding
based on the topic proportions using the <code>umap</code> function.
This projects cells into a low-dimensional space that preserves
topic-driven structure. By default the embedding is saved as
<code>mTopic_umap</code> reduction.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/umap.html">umap</a></span><span class="op">(</span><span class="va">pbmc</span>, n_components <span class="op">=</span> <span class="fl">2</span>, min_dist <span class="op">=</span> <span class="fl">0.1</span>, n_neighbors <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: No assay specified, setting assay as RNA by default.</span></span>
<span><span class="va">pbmc</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 147123 features across 7349 samples within 3 assays </span></span>
<span><span class="co">#&gt; Active assay: atac (136915 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  2 other assays present: rna, prot</span></span>
<span><span class="co">#&gt;  2 dimensional reductions calculated: mTopic, mTopic_umap</span></span></code></pre></div>
<p>The <code>umap</code> function is designed to create an embedding
based on the topic proportions only, rather than, for example, PCA. This
can also be achieved using a standard approach from the Seurat
package.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">pbmc</span>, reduction <span class="op">=</span> <span class="st">'mTopic'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p>Now you can explore the topic-cell distributions by visualizing
individual topic proportions or dominant topics using functions like
<code>plot_topics</code> and <code>plot_dominant_topics</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_topics.html">plot_topics</a></span><span class="op">(</span><span class="va">pbmc</span>, x <span class="op">=</span> <span class="st">'mTopic_umap'</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial3_plot1.png" width="100%"></p>
<p>Topic distributions can also be visualized with the Seurat
<code><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot()</a></code> function:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">pbmc</span>, order <span class="op">=</span> <span class="cn">TRUE</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"topic_"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span> <span class="op">*</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_equal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial-3-human-pbmcs_files/figure-html/unnamed-chunk-11-1.png" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">'topic_1'</span>  <span class="op">=</span> <span class="st">'#CDBA96'</span>, <span class="st">'topic_2'</span>  <span class="op">=</span> <span class="st">'#000080'</span>, <span class="st">'topic_3'</span>  <span class="op">=</span> <span class="st">'#FFFF00'</span>, </span>
<span>  <span class="st">'topic_4'</span>  <span class="op">=</span> <span class="st">'#9ECAE1'</span>, <span class="st">'topic_5'</span>  <span class="op">=</span> <span class="st">'#189E7F'</span>, <span class="st">'topic_6'</span>  <span class="op">=</span> <span class="st">'#FFF8DC'</span>,</span>
<span>  <span class="st">'topic_7'</span>  <span class="op">=</span> <span class="st">'#642915'</span>, <span class="st">'topic_8'</span>  <span class="op">=</span> <span class="st">'#0000FF'</span>, <span class="st">'topic_9'</span>  <span class="op">=</span> <span class="st">'#BEBEBE'</span>,</span>
<span>  <span class="st">'topic_10'</span> <span class="op">=</span> <span class="st">'#4292C6'</span>, <span class="st">'topic_11'</span> <span class="op">=</span> <span class="st">'#CD0000'</span>, <span class="st">'topic_12'</span> <span class="op">=</span> <span class="st">'#58508D'</span>, </span>
<span>  <span class="st">'topic_13'</span> <span class="op">=</span> <span class="st">'#006400'</span>, <span class="st">'topic_14'</span> <span class="op">=</span> <span class="st">'#10D894'</span>, <span class="st">'topic_15'</span> <span class="op">=</span> <span class="st">'#666666'</span>, </span>
<span>  <span class="st">'topic_16'</span> <span class="op">=</span> <span class="st">'#800080'</span>, <span class="st">'topic_17'</span> <span class="op">=</span> <span class="st">'#FF00FF'</span>, <span class="st">'topic_18'</span> <span class="op">=</span> <span class="st">'#FFC0CB'</span>,</span>
<span>  <span class="st">'topic_19'</span> <span class="op">=</span> <span class="st">'#BEBEBE'</span>, <span class="st">'topic_20'</span> <span class="op">=</span> <span class="st">'#FFD700'</span>, <span class="st">'topic_21'</span> <span class="op">=</span> <span class="st">'#BEBEBE'</span>,</span>
<span>  <span class="st">'topic_22'</span> <span class="op">=</span> <span class="st">'#00FFFF'</span>, <span class="st">'topic_23'</span> <span class="op">=</span> <span class="st">'#CD853F'</span>, <span class="st">'topic_24'</span> <span class="op">=</span> <span class="st">'#DDA0DD'</span>,</span>
<span>  <span class="st">'topic_25'</span> <span class="op">=</span> <span class="st">'#000000'</span>, <span class="st">'topic_26'</span> <span class="op">=</span> <span class="st">'#551A8B'</span>, <span class="st">'topic_27'</span> <span class="op">=</span> <span class="st">'#00CD00'</span>, </span>
<span>  <span class="st">'topic_28'</span> <span class="op">=</span> <span class="st">'#F4A460'</span>, <span class="st">'topic_29'</span> <span class="op">=</span> <span class="st">'#00FF7F'</span>, <span class="st">'topic_30'</span> <span class="op">=</span> <span class="st">'#FFBBFF'</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_dominant_topics.html">plot_dominant_topics</a></span><span class="op">(</span><span class="va">pbmc</span>, x <span class="op">=</span> <span class="st">'mTopic_umap'</span>, palette <span class="op">=</span> <span class="va">palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial3_plot2.png" width="100%"></p>
<p>To further interpret the results of the MTM, you can plot feature
signatures for each modality. Feature signatures represent the
importance or contribution of specific features — genes, chromatin
accessibility regions, or proteins—to each topic.</p>
<p>Use <code>plot_signatures</code> to visualize the top features per
topic for any selected modality, helping to identify key markers that
define each topic’s biological identity.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_signatures.html">plot_signatures</a></span><span class="op">(</span><span class="va">pbmc</span>, mod<span class="op">=</span><span class="st">'atac'</span>, n_top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial3_plot3.png" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_signatures.html">plot_signatures</a></span><span class="op">(</span><span class="va">pbmc</span>, mod <span class="op">=</span> <span class="st">'rna'</span>, n_top <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial3_plot4.png" width="100%"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_signatures.html">plot_signatures</a></span><span class="op">(</span><span class="va">pbmc</span>, mod <span class="op">=</span> <span class="st">'prot'</span>, n_top <span class="op">=</span> <span class="fl">10</span>, figsize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial3_plot5.png" width="100%"></p>
<p>To conclude the tutorial, save the <code>pbmc</code> object with the
trained parameters to a file. This ensures the results can be reloaded
later for further analysis or visualization.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">pbmc</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="st">'Human_PBMC_ATAC_RNA_Protein_trained.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Damian Panas, Piotr Rutkowski.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
