<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tutorial 2: Human Tonsil (RNA + Protein Epitopes) • mTopic</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial 2: Human Tonsil (RNA + Protein Epitopes)">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mTopic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tutorial-1-data-preparation.html">Data Preparation</a>
    </li>
    <li>
      <a href="../articles/tutorial-2-human-tonsil.html">Human Tonsil (RNA + Protein Epitopes)</a>
    </li>
    <li>
      <a href="../articles/tutorial-3-human-pbmcs.html">Human PBMCs (ATAC + RNA + Protein Epitopes)</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">API</a>
</li>
<li>
  <a href="https://github.com/TabakaLab/mTopic" class="external-link">Python Package</a>
</li>
<li>
  <a href="https://github.com/TabakaLab/mTopic/tree/main/dashboard" class="external-link">Interactive Dashboard</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/TabakaLab/mTopicR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Tutorial 2: Human Tonsil (RNA + Protein
Epitopes)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TabakaLab/mTopicR/blob/HEAD/vignettes/tutorial-2-human-tonsil.Rmd" class="external-link"><code>vignettes/tutorial-2-human-tonsil.Rmd</code></a></small>
      <div class="hidden name"><code>tutorial-2-human-tonsil.Rmd</code></div>

    </div>

    
    
<p>Welcome to the tutorial on using the <code>mTopic</code> package for
spatial multimodal topic modeling of human tonsils dataset. This dataset
offers the following key features:</p>
<ul>
<li>
<strong>Multimodal measurements.</strong> Combines RNA and protein
data, enabling integrated gene expression analysis and surface protein
abundance analysis.</li>
<li>
<strong>Spatial context.</strong> Includes spatial coordinates for
each cell or spot, allowing spatially informed analysis and
visualization of tissue organization.</li>
</ul>
<p>In this tutorial, we will walk through the following steps:</p>
<ul>
<li>Scaling and normalizing the data.</li>
<li>Applying spatial multimodal topic modeling to identify distinct cell
populations and explore their functional roles.</li>
<li>Visualizing the results to gain insight into the spatial
distribution of topics and cell types within the tissue.</li>
</ul>
<p>Let’s begin by importing the required libraries for this analysis.
The filtered training data is available at <a href="https://zenodo.org/records/15431784" class="external-link">Zenodo</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TabakaLab/mTopicR" class="external-link">mTopic</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">dataset_name</span> <span class="op">&lt;-</span> <span class="st">"Human_Tonsil_RNA_Protein_filtered.rds"</span></span>
<span><span class="va">working_dir</span> <span class="op">&lt;-</span> <span class="st">'~/projects/datasets/'</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="va">dataset_name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span></span>
<span>    url <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/15431784/files/"</span>, <span class="va">dataset_name</span>, <span class="st">"?download=1"</span><span class="op">)</span>, </span>
<span>    destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="va">dataset_name</span><span class="op">)</span>, </span>
<span>    mode <span class="op">=</span> <span class="st">"wb"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="section level2">
<h2 id="spatial-multimodal-topic-modeling">Spatial Multimodal Topic Modeling<a class="anchor" aria-label="anchor" href="#spatial-multimodal-topic-modeling"></a>
</h2>
<p>Load the prefiltered <code>Seurat</code> object with the human
tonsils dataset. The dataset comprises 4,194 spatial spots and two
modalities, <code>rna</code> (gene expression) and <code>prot</code>
(protein levels).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="va">dataset_name</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Before we train the spatial Multimodal Topic Model
(<code>sMTM</code>), it is essential to preprocess the data to improve
the model’s ability to identify meaningful patterns across
modalities.</p>
<p>To ensure comparability between RNA and protein data, we apply the
following normalization and scaling steps:</p>
<ul>
<li>
<strong>TF-IDF transformation for RNA (<code>tfidf</code>).</strong>
Adjusts raw gene expression counts by balancing feature frequency and
importance, emphasizing rare but informative genes.</li>
<li>
<strong>CLR normalization for protein (<code>clr</code>).</strong>
Corrects compositional biases by normalizing protein counts across cells
using the Centered Log Ratio method.</li>
<li>
<strong>Scaling across modalities
(<code>scale_counts</code>).</strong> Linearly scales counts to ensure
all modalities contribute equally during topic modeling, preventing one
from dominating the analysis.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tfidf.html">tfidf</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clr.html">clr</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'prot'</span><span class="op">)</span></span>
<span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scale_counts.html">scale_counts</a></span><span class="op">(</span><span class="va">ton</span><span class="op">)</span></span></code></pre></div>
<p>Now that the data is preprocessed, we can train the spatial
Multimodal Topic Model (<code>sMTM</code>). This model identifies
coordinated patterns across modalities while incorporating spatial
information. It captures co-expression of genes and proteins, revealing
distinct cell populations and their functional states.</p>
<ul>
<li><p><strong>Initialize the model</strong>. Create an instance of the
<code>MTM</code> class, specifying the number of topics
(<code>n_topics</code>) and other parameters. We use 19 topics for this
tutorial for a compact but expressive model.</p></li>
<li><p><strong>Train the model</strong>. Fit the model using Variational
Inference (<code>VI</code>). This iterative process updates the model
parameters to explain the observed data best. While training time
depends on dataset size, <code>sMTM</code> is optimized for
scalability.</p></li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">smtm_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sMTM.html">sMTM</a></span><span class="op">(</span><span class="va">ton</span>, n_topics <span class="op">=</span> <span class="fl">19</span>, radius <span class="op">=</span> <span class="fl">0.02</span>, n_jobs <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">smtm_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/VI.html">VI</a></span><span class="op">(</span><span class="va">smtm_model</span>, n_iter <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p>After training, export the learned parameters to the Seurat object
the using <code>export_params</code> function:</p>
<ul>
<li><p>Topic proportions (<code>gamma</code> saved to
<code>tonsils@reductions$mTopic</code>). Represent the distribution of
topics within each cell or spot. These proportions help identify which
topics are active in different tissue regions.</p></li>
<li><p>Topic-feature distributions (<code>lambda</code> saved to
<code>tonsils[[modality]]@meta.data</code>). Indicate the importance of
each feature (gene or protein) for each topic. These distributions help
interpret the biological meaning of the topics.</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/export_params.html">export_params</a></span><span class="op">(</span><span class="va">smtm_model</span>, <span class="va">ton</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: No assay specified, setting assay as RNA by default.</span></span>
<span><span class="va">ton</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 5028 features across 4194 samples within 2 assays </span></span>
<span><span class="co">#&gt; Active assay: prot (28 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  1 other assay present: rna</span></span>
<span><span class="co">#&gt;  2 dimensional reductions calculated: Spatial, mTopic</span></span></code></pre></div>
<p>The <code>export_params</code>. function not only exports model
parameters but can also filter out insignificant topics based on their
overall contribution. Topic filtering: Set
<code>filter_topics = TRUE</code> to automatically remove topics with
low overall activity. The <code>filter_threshold</code> parameter
controls the minimum average topic proportion required to retain a
topic.</p>
<p>This helps remove spurious or noise-driven topics, simplifying
downstream interpretation and visualization.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_filter_topics.html">plot_filter_topics</a></span><span class="op">(</span><span class="va">smtm_model</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot1.png" width="100%"></p>
<p>By preprocessing the data, training the <code>sMTM</code> model, and
exporting the learned parameters, we have set the stage for a
comprehensive analysis of cellular heterogeneity.</p>
</div>
<div class="section level2">
<h2 id="visualizing-topic-proportions">Visualizing Topic Proportions<a class="anchor" aria-label="anchor" href="#visualizing-topic-proportions"></a>
</h2>
<p>Visualizing topic distribution across cells or spatial spots is key
in interpreting topic modeling results. These proportions reflect the
contribution of each topic to each cell. They can reveal spatially
organized cell states, types, or biological processes.</p>
<p>To visualize topic proportions, use the <code>plot_topics</code>
function to generate scatter plots where each cell or spot is colored
according to the value of a selected topic. This reveals spatial
patterns and gradients that help interpret biological variation within
the tissue.</p>
<p>For example, if a topic captures a specific cell type, the plot will
highlight regions enriched in that population.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_topics.html">plot_topics</a></span><span class="op">(</span><span class="va">ton</span>, x <span class="op">=</span> <span class="st">'Spatial'</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot2.png" width="100%"></p>
<p>To visualize overall trends in topic distributions, use the
<code>plot_dominant_topics</code> function. This function assigns each
cell or spot to its most dominant topic — the one with the highest
proportion — and colors it accordingly.</p>
<p>The resulting plot provides a global overview of topic dominance
across the tissue, helping you quickly identify regions enriched in
specific topics. These regions may correspond to distinct cell types,
tissue structures, or gradients of biological activity.</p>
<p>This visualization is handy for detecting the tissue’s spatial
domains and functional zones.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">'topic_1'</span>  <span class="op">=</span> <span class="st">'#F4A460'</span>, <span class="st">'topic_2'</span>  <span class="op">=</span> <span class="st">'#00CD00'</span>, <span class="st">'topic_3'</span>  <span class="op">=</span> <span class="st">'#00FF7F'</span>, </span>
<span>  <span class="st">'topic_4'</span>  <span class="op">=</span> <span class="st">'#FFC0CB'</span>, <span class="st">'topic_5'</span>  <span class="op">=</span> <span class="st">'#0000CD'</span>, <span class="st">'topic_6'</span>  <span class="op">=</span> <span class="st">'#000000'</span>,</span>
<span>  <span class="st">'topic_7'</span>  <span class="op">=</span> <span class="st">'#BEBEBE'</span>, <span class="st">'topic_8'</span>  <span class="op">=</span> <span class="st">'#FF6347'</span>, <span class="st">'topic_9'</span>  <span class="op">=</span> <span class="st">'#20B2AA'</span>,</span>
<span>  <span class="st">'topic_10'</span> <span class="op">=</span> <span class="st">'#FF69B4'</span>, <span class="st">'topic_11'</span> <span class="op">=</span> <span class="st">'#CDBA96'</span>, <span class="st">'topic_12'</span> <span class="op">=</span> <span class="st">'#BEBEBE'</span>,</span>
<span>  <span class="st">'topic_13'</span> <span class="op">=</span> <span class="st">'#EE3A8C'</span>, <span class="st">'topic_14'</span> <span class="op">=</span> <span class="st">'#006400'</span>, <span class="st">'topic_15'</span> <span class="op">=</span> <span class="st">'#FFFF00'</span>,</span>
<span>  <span class="st">'topic_16'</span> <span class="op">=</span> <span class="st">'#CD853F'</span>, <span class="st">'topic_17'</span> <span class="op">=</span> <span class="st">'#FFF8DC'</span>, <span class="st">'topic_18'</span> <span class="op">=</span> <span class="st">'#EE82EE'</span>,</span>
<span>  <span class="st">'topic_19'</span> <span class="op">=</span> <span class="st">'#8B2252'</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_dominant_topics.html">plot_dominant_topics</a></span><span class="op">(</span><span class="va">ton</span>, x <span class="op">=</span> <span class="st">'Spatial'</span>, marker <span class="op">=</span> <span class="st">'h'</span>, s <span class="op">=</span> <span class="fl">30</span>, palette <span class="op">=</span> <span class="va">palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot3.png" width="100%"></p>
<p>Alternatively, you can explore topic proportions in more detail using
pie chart scatter plots with <code>plot_scatter_pie</code>. This
function visualizes the complete topic composition of each cell or spot
as a pie chart.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_scatter_pie.html">plot_scatter_pie</a></span><span class="op">(</span><span class="va">ton</span>, x <span class="op">=</span> <span class="st">'Spatial'</span>, radius <span class="op">=</span> <span class="fl">0.0073</span>, palette <span class="op">=</span> <span class="va">palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot4.png" width="100%"></p>
<p>Rendering may take time, especially for large datasets. To improve
performance and focus on a specific region, you can limit the number of
visualized spots using the <code>xrange</code> and <code>yrange</code>
parameters (default: <code>c(0, 1)</code>), which define the fraction of
the spatial extent to display.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_scatter_pie.html">plot_scatter_pie</a></span><span class="op">(</span><span class="va">ton</span>, x <span class="op">=</span> <span class="st">'Spatial'</span>, radius <span class="op">=</span> <span class="fl">0.0073</span>, palette <span class="op">=</span> <span class="va">palette</span>, xrange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.6</span><span class="op">)</span>, yrange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.35</span>, <span class="fl">0.65</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot5.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="visualizing-feature-signatures">Visualizing Feature Signatures<a class="anchor" aria-label="anchor" href="#visualizing-feature-signatures"></a>
</h2>
<p>To interpret the results of the <code>sMTM</code> model, it is
important to examine the feature signatures associated with each topic.
Identifying the most relevant features — genes or proteins — for each
topic provides insight into the biological identity and function of the
inferred cell populations or processes.</p>
<p>Use the <code>plot_signatures</code> function to visualize the top
features per topic. This function generates a set of plots, each showing
the most significant features ranked by their importance for a given
topic.</p>
<p>These visualizations help reveal which molecular markers distinguish
topics, aiding in biological interpretation and annotation of the
results.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_signatures.html">plot_signatures</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'rna'</span>, n_top <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot6.png" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_signatures.html">plot_signatures</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'prot'</span>, n_top <span class="op">=</span> <span class="fl">10</span>, figsize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot7.png" width="100%"></p>
<p>To better understand the spatial relevance of topic signatures and
validate their biological specificity, you can visualize feature
z-scores. A z-score indicates how much a feature’s expression in a given
cell deviates from the mean, normalized by standard deviation. This
highlights significantly up- or downregulated features in specific
regions or cell populations.</p>
<p>Use <code>zscores</code> to compute modality-specific z-scores, and
<code>plot_zscores</code> to visualize their spatial distribution across
the tissue. These plots help identify spatially localized or diffuse
topics, revealing potential functional niches or gradients.</p>
<p>In the example below, we compute z-scores for the top 20 genes and
top 3 proteins per topic to explore their spatial expression
patterns.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/zscores.html">zscores</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'rna'</span>, n_top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_zscores.html">plot_zscores</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'rna'</span>, x <span class="op">=</span> <span class="st">'Spatial'</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot8.png" width="100%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ton</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/zscores.html">zscores</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'prot'</span>, n_top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_zscores.html">plot_zscores</a></span><span class="op">(</span><span class="va">ton</span>, mod <span class="op">=</span> <span class="st">'prot'</span>, x <span class="op">=</span> <span class="st">'Spatial'</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot9.png" width="100%"></p>
<p>It is helpful to visualize the correlation between z-scores and topic
proportions across cells to assess how well feature signatures align
with inferred topic distributions.</p>
<p>Use the <code>plot_corr_heatmap</code> function to generate a heatmap
of correlations between z-scores (for top features) and topic
proportions. Each cell in the heatmap represents the correlation
strength for a feature-topic pair.</p>
<p>High values along the diagonal indicate that the selected features
strongly correspond to their respective topics. This confirms that the
topic model has captured biologically meaningful patterns.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_corr_heatmap.html">plot_corr_heatmap</a></span><span class="op">(</span>data1<span class="op">=</span><span class="va">ton</span><span class="op">@</span><span class="va">reductions</span><span class="op">$</span><span class="va">mTopic</span><span class="op">@</span><span class="va">cell.embeddings</span>, label1 <span class="op">=</span> <span class="st">'sMTM topics'</span>, </span>
<span>                  data2<span class="op">=</span><span class="va">ton</span><span class="op">[[</span><span class="st">'rna'</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">zscores</span>, label2 <span class="op">=</span> <span class="st">'RNA z-scores'</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot10.png" width="100%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_corr_heatmap.html">plot_corr_heatmap</a></span><span class="op">(</span>data1<span class="op">=</span><span class="va">ton</span><span class="op">@</span><span class="va">reductions</span><span class="op">$</span><span class="va">mTopic</span><span class="op">@</span><span class="va">cell.embeddings</span>, label1 <span class="op">=</span> <span class="st">'sMTM topics'</span>, </span>
<span>                  data2<span class="op">=</span><span class="va">ton</span><span class="op">[[</span><span class="st">'prot'</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">zscores</span>, label2 <span class="op">=</span> <span class="st">'Protein z-scores'</span><span class="op">)</span></span></code></pre></div>
<p><img src="tutorial2_plot11.png" width="100%"></p>
<p>This concludes the example application of <code>mTopic</code> for
modeling spatial multimodal single-cell data, demonstrated using the
human tonsil dataset. We have walked through preprocessing, topic
modeling, and result interpretation, highlighting how
<code>mTopic</code> enables integrated analysis across modalities with
spatial context.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">ton</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">working_dir</span>, <span class="st">'Human_Tonsil_RNA_Protein_trained.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Damian Panas, Piotr Rutkowski.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
